### PR TITLE
Support jetty 11

### PR DESCRIPTION
**Pull request by xu1009**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. https://github.com/apache/skywalking/issues/10707
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ metrics based on the tracing data.
   * [Struts2 MVC](http://struts.apache.org/)  2.3.x -> 2.5.x
   * [Resin](https://www.caucho.com/resin-4.0/) 3 (Optional¹)
   * [Resin](https://www.caucho.com/resin-4.0/) 4 (Optional¹)
-  * [Jetty Server](http://www.eclipse.org/jetty/) 9
+  * [Jetty Server](http://www.eclipse.org/jetty/) 9.x -> 11.x
```
[the remark this answers is not in the document]
Comment (xu1009) [reply docs/en/setup/service-agent/java-agent/Supported-list.md:17]: i ave tested 10.x server and client locllly，i will add 10.x scenario test，10.x server is same with 9.x, jetty client is same with 11.x，and it requires jdk11+

File: apm-sniffer/apm-sdk-plugin/jetty-plugin/jetty-server-11.x-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>apm-jetty-server-11.x-plugin</artifactId>
+    <packaging>jar</packaging>
+
+    <name>jetty-server-11.x-plugin</name>
+    <url>http://maven.apache.org</url>
+
+    <properties>
+        <jetty-server.version>11.0.15</jetty-server.version>
```
[the remark this answers is not in the document]
Comment (xu1009) [reply apm-sniffer/apm-sdk-plugin/jetty-plugin/jetty-server-11.x-plugin/pom.xml:34]: not get your mean, why move it to the optional plugins

[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/jetty-plugin/jetty-server-11.x-plugin/pom.xml:34]: I don't think we need this to be optional, as it doesn't have conflict with others.

**Review by lujiajing1126 — APPROVED** [review]
LGTM

**Review by wu-sheng — APPROVED** [review]
LGTM

**Pull request merged**